packages/icons/src/Icon.tsx

```diff
@@ -52,10 +52,20 @@ export const Icon = React.forwardRef(function Icon(
 
     const [iconStyle, icon] = getIconStyle(propIconStyle, propIcon);
     const url = getIconAssetURL(context, iconStyle, icon);
+    const [loaded, setLoaded] = React.useState(false);
+
+    React.useEffect(() => {
+        const img = new Image();
+        img.src = url;
+        img.onload = () => {
+            setLoaded(true);
+        };
+    }, [url]);
```

Comment:
Why do we do this? can you add a quick comment 

It looks like it's for visual tests. Can it impact perfs on prod? I guess the browser only loads it once?

Reply:
Yeah it's for visual test, I think there is no impact on performance. Because it's in cache. But it still a double render of icon, I will see if I can do better.

Reply:
Before taking the screenshot, can we wait for all requests to be done?

Reply:
It's not recommended but yeah we could try that.